core/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java

```diff
@@ -118,28 +120,58 @@ public MatchQueryBuilder analyzer(String analyzer) {
         return this;
     }
 
-    /**
-     * Set the phrase slop if evaluated to a phrase query type.
-     */
+    /** Get the analyzer to use, if previously set, otherwise <tt>null</tt> */
+    public String analyzer() {
+        return this.analyzer;
+    }
+
+    /** Sets a slop factor for phrase queries */
     public MatchQueryBuilder slop(int slop) {
+        if (slop < 0 ) {
+            throw new IllegalArgumentException("No negative slop allowed.");
+        }
         this.slop = slop;
         return this;
     }
 
-    /**
-     * Sets the fuzziness used when evaluated to a fuzzy query type. Defaults to "AUTO".
-     */
+    /** Get the slop factor for phrase queries. Returns <tt>null</tt> if not previously set. */
+    public Integer slop() {
```

Comment:
Changed this to `int 0` as default. Thats what MatchQuery does internally if not set.